### PR TITLE
Add email unsubscribe UI and logic

### LINKED ISSUE #4465: As a user who is annoyed by Gitcoin's emails, I want a 'surgical unsubscribe' button so I can unsubscribe from one email in one click.
### User Story

[comment]: # (As a <user type>, I want to <task> so that <goal>.)
As a user who is annoyed by Gitcoin's emails, I want a 'surgical unsubscribe' button so I can unsubscribe from one email in one click.

### Why Is this Needed

[comment]: # (Describe the problem and why this task is needed. Provide description of the current state, what you would like to happen, and what actually happen)
*Summary*:
Right now you
1. can't unsubscribe in one click
2. are taken to settings/email when you click unsubscribe, tempting you to unsubscribe from ALL emails.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*:
feature

### Current Behavior
[comment]: # (Describe what actually happened.)
clicking 'unsubscribe' in your email will take you to settings/email when you click unsubscribe, tempting you to unsubscribe from ALL emails.

### Expected Behavior
[comment]: # (Describe what you expected to happen.)
when a user clicks unsubscribe in any email, they are taken to a page that looks more like [this](http://bits.owocki.com/64f88bb1ece4/Image%202019-05-22%20at%203.31.48%20PM.png) -- basically lets you know you've been unsubscribed from this ONE email

### Definition of Done
[comment]: # (Any other information that would be useful, bullets are helpful.)
see above

### Data 
Decrease in % of completely unsubscribed users

### Reach
All customers that see this email

### Impact
Decrease Unsubscribes, High Impact 

### Confidence 
80% 

### Additional Information
[comment]: # (Any other information that would be useful, content, screenshots, etc.)

### PR DESCRIPTION
**Pull request by iamonuwa**
##### Description

1. Unsubscribe user from a particular email type instead of all.
2. Updated the emails to pass the email_type as part of the requests on the unsubscribe email
3. Redirect to unsubscribe email page if successful
4. Ensure that only valid email_types are processed. Others redirect to settings page

##### Refers/Fixes
#4465 

##### Testing

Click on the _unsubscribe_ link at the footer of an email from Gitcoin.

### CODE REVIEW

**Comment by owocki** [comment]
@danlipert i can review this if u guys want

**Comment by octavioamu** [comment]
@iamonuwa why you didn't used https://github.com/gitcoinco/web/pull/4653#issuecomment-534347440 ?

**Comment by iamonuwa** [comment]
> @iamonuwa why you didn't used [#4653 (comment)](https://github.com/gitcoinco/web/pull/4653#issuecomment-534347440) ? 187kb vs 51kb

Exported same file from figma as jpg. But will update it with your

**Comment by iamonuwa** [comment]
@owocki any update on this one? We can chat on slack 🙈

File: app/retail/emails.py [hunk 1]
```diff
@@ -479,6 +487,7 @@ def render_new_bounty(to_email, bounties, old_bounties, offset=3):
         'subscriber': sub,
         'keywords': ",".join(sub.keywords),
         'email_style': email_style,
+		'email_type': 'new_bounty_notifications'
```
Comment (owocki) [inline app/retail/emails.py]: all of these email type fields have a mix of tabs/spaces. shuld just be 4 spaces per tab.

File: app/marketing/views.py [hunk 2]
```diff
@@ -272,6 +272,42 @@ def email_settings(request, key):
     email = ''
     level = ''
     msg = ''
+    email_types = {
+        'welcome_mail': 'Welcome Email',
```
Comment (owocki) [inline app/marketing/views.py]: this already exists in `retail.emails.MARKETING_EMAILS`  - it should be DRYd up

File: app/retail/templates/email_unsubscribed.html [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program. If not, see <http://www.gnu.org/licenses/>.
+{% endcomment %}
+{% load i18n static %}
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    {% include 'shared/head.html' %}
```
Comment (owocki) [inline app/retail/templates/email_unsubscribed.html]: hmm this logo looks messed up http://bits.owocki.com/5022c094c65c/Screen%20Shot%202019-10-14%20at%201.06.12%20PM.png

File: app/retail/templates/email_unsubscribed.html [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program. If not, see <http://www.gnu.org/licenses/>.
+{% endcomment %}
```
Comment (owocki) [inline app/retail/templates/email_unsubscribed.html]: when i go to this page it advertises that i've been unsubscribed but doesnt actually flip the switch in the DB.  http://bits.owocki.com/8d5b49fe9d37/Screen%20Recording%202019-10-14%20at%2001.07%20PM.gif

if were saying its a one click unsubscribe then it probably should be

**Comment by owocki** [comment]
jusft left a handful of comments

**Comment by owocki** [comment]
@iamonuwa did u see my comments

File: app/marketing/views.py (hunk 2 above)
Comment (iamonuwa) [reply app/marketing/views.py]: I don't understand this. Care to elaborate? `retail.emails.MARKETING_EMAILS.welcome_mail` you mean?

Comment (owocki) [reply app/marketing/views.py]: DRY = dont' repeat yourself.  we dont need another `email_types` definition dict when it already exists on `retail.emails.MARKETING_EMAILS`

Comment (iamonuwa) [reply app/marketing/views.py]: ```retail.emails.MARKETING_EMAILS``` is a list.

File: app/retail/templates/email_unsubscribed.html (hunk 3 above)
Comment (iamonuwa) [reply app/retail/templates/email_unsubscribed.html]: The logo came from the figma design. I exported it from there

File: app/marketing/views.py (hunk 2 above)
Comment (owocki) [reply app/marketing/views.py]: it still contains the same information that you're re-declaring here and should be DRY'd..  mutate one data structure into the other if you need to

File: app/retail/templates/email_unsubscribed.html (hunk 3 above)
Comment (owocki) [reply app/retail/templates/email_unsubscribed.html]: here is a better logo http://bits.owocki.com/ec1d9a5520d0/gitcoin_logo_white%20copy%203.png

**Comment by owocki** [comment]
ok since @iamchrissmith went MIA; i've taken the liberty of addressing my own coments so this can finally go live. 9d4d96371560121894a3b792dcc9019f771445a9

**Comment by owocki** [comment]
ok since @iamonuwa  went MIA; i've taken the liberty of addressing my own comnents so this can finally go live. 9d4d96371560121894a3b792dcc9019f771445a9

**Pull request merged**